src/__tests__/permissionSelectors-test.js

```diff
@@ -0,0 +1,130 @@
+/* @flow strict-local */
+import invariant from 'invariant';
+
+import { tryGetActiveAccountState } from '../selectors';
+import {
+  Role,
+  type RoleT,
+  CreateWebPublicStreamPolicy,
+  type CreateWebPublicStreamPolicyT,
+} from '../api/permissionsTypes';
+import { getCanCreateWebPublicStreams } from '../permissionSelectors';
+import rootReducer from '../boot/reducers';
+import { EVENT_USER_UPDATE } from '../actionConstants';
+import * as eg from './lib/exampleData';
+
+describe('getCanCreateWebPublicStreams', () => {
+  const { AdminOrAbove, ModeratorOrAbove, Nobody, OwnerOnly } = CreateWebPublicStreamPolicy;
+  const { Owner, Admin, Moderator, Member, Guest } = Role;
+
+  test('returns false when webPublicStreamsEnabled is false', () => {
+    const globalState = [
+      {
+        type: EVENT_USER_UPDATE,
+        id: 0,
+        userId: eg.selfUser.user_id,
+        person: { user_id: eg.selfUser.user_id, role: Owner },
+      },
+    ].reduce(
+      rootReducer,
+      eg.reduxStatePlus({
+        realm: {
+          ...eg.plusReduxState.realm,
+          webPublicStreamsEnabled: false,
+          enableSpectatorAccess: true,
+          createWebPublicStreamPolicy: ModeratorOrAbove,
+        },
+      }),
```

Comment:
I think it might be nice to use a completely empty initial state here, then apply changes like `createWebPublicStreamPolicy` in a minimally modified `eg.action.register_complete`. The problem is that `eg.reduxStatePlus` gives us boring things that we need, like putting `selfUser` in `state.users`, that we don't get from `eg.action.register_complete`.

What do we think about changing `eg.action.register_complete`'s interface such that it would make an initial state look more like `eg.plusReduxState`? We could even construct `eg.plusReduxState` using this modified `eg.action.register_complete`.

Reply:
> to change `eg.action.register_complete`'s interface such that it would make a blank state look more like `eg.plusReduxState`? We could even construct `eg.plusReduxState` using this modified `eg.action.register_complete`.

Yeah, that sounds like a good direction.

I'm not sure if we have a use case that needs a "minimal" version of the action, which is what that's currently documented as. Would have to look at the use sites of it to see. If we do, we could potentially have both -- first the minimal, then also one that's defined by starting from that and adding things we need to add.